### DISCUSSION TITLE
dlaczego-rss-nadal-ma-znaczenie-w-2021/

### DISCUSSION THREAD

**Discussion by giscus** [idarek/dariusz.wieckiewicz.org#125] [General]
# dlaczego-rss-nadal-ma-znaczenie-w-2021/

RSS to technologia umożliwiająca subskrypcję witryn internetowych i blogów oraz otrzymywanie aktualizacji po opublikowaniu nowej treści. Może to być świetny sposób na bycie na bieżąco z ulubionymi witrynami bez konieczności odwiedzania ich każdego dnia. RSS to nadal cenne narzędzie w 2021 r. i może być świetnym sposobem na oszczędzanie czasu i zachowanie porządku.

https://dariusz.wieckiewicz.org/dlaczego-rss-nadal-ma-znaczenie-w-2021/

**Comment by xpil** [comment]
Feedly trzyma twoje dane w chmurze, więc jak się kiedyś wyprowadzą, to zostaniesz z niczym. Chyba, że sobie wyeksportujesz swoje kanały do pliku OPML, ale to trzeba (A) wiedzieć, i (B) robić regularnie. Aha, no i (C) tracisz historię przeczytanych / ulubionych artykułów. Czyli tak czy siak dupa z tyłu. Podobnie ma się sprawa z Inoreader, NewsBlur i innymi czytnikami w chmurze.

Dlatego najlepszym (według mnie) rozwiązaniem jest albo (A) mieć lokalną aplikację (Thunderbird, Outlook, QuiteRSS lub podobną), która trzyma wszystko na lokalnej maszynie (lub w chmurze dostawcy + na lokalnej maszynie), albo (B) hostować czytnik RSS w postaci rozwiązania klient - serwer na publicznie dostępnym hoście, nad którym mamy pełną kontrolę.

Prywatnie wolę to ostatnie rozwiązanie, ponieważ w ten sposób mam dostęp do swoich kanałów RSS z każdego miejsca na Ziemi, o ile jest tam internet z przeglądarką. Kopie zapasowe robią się automatycznie raz dziennie, w chmurę Dropbox. Wszystko po prostu działa. Nie licząc odrobiny wysiłku umysłowego i czasu na samym początku, żeby to wszystko poustawiać, cała reszta odbywa się w stu procentach bezobsługowo.

Konkretnie to używam Miniflux, bo jest natywnym pakietem Debiana (a więc aktualizuje się razem z systemem), a do tego jest cudnie minimalistyczny, bardzo dobrze pokumany i super kompatybilny z pierdylionem usług zewnętrznych. Pisałem o tym nawet niedawno u siebie: https://xpil.eu/miniflux-5-miesiecy-pozniej/. Ale są alternatywy (tt-rss, feedbin i mnóstwo innych), więc jak ktoś ma odrobinę wiedzy i maszynę z publicznym IP, ma z czego wybierać.

**Reply by xpil** [reply]
P.S. Jedną z nielicznych wad technologii RSS jest możliwość manipulowania datami. Musiałem kiedyś pozbyć się jednego kanału z czytnika, bo miał ustawioną datę publikacji wpisu z reklamą kanału na trzy lata do przodu, przez co zawsze wyskakiwał na samej górze na liście postów do przeczytania. Podobnie zresztą tutaj - ten wpis jest z 2021 roku, a ja go zobaczyłem u siebie dzisiaj, w 2024 (czyli ktoś lub coś gdzieś zmanipulowało datę ostatniej modyfikacji tego wpisu i mój czytnik uznał, że to funkiel nówka nieśmigana). Że to stary wpis zorientowałem się poniewczasie.

**Reply by idarek** [reply]
W moim przypadku to jest to, że wpis ten, mimo że z 2022, oryginalnie był tylko po Angielsku a teraz jest przedłumaczony na Polski. RSS ma czasem pamięć taką, że mimo daty ustawionej wstecz, gdy URL jest nowy, wówczas się pojawi jako nowy, mimo, że z przed kilku lat.

ps. Daty ustawione poprawnie, 2021/2022 :)

**Reply by xpil** [reply]
Nie zmienia to jednak w żaden sposób mojej opinii o czytnikach RSS trzymających dane w obcej chmurze :)

**Reply by spexpl** [reply]
@xpil (ad. daty) to dużo zależy od tego jak dany CMS ma zachowywać się przy aktualizacji treści. Bo WordPress domyślnie nie re-publikuje wpisy w trakcie jego edycji. Trzeba to obchodzić na około poprzez zaplanowaną publikacje (z harmonogramu) wpisu - wtedy wpis ponownie trafia do feeda. 

Widocznie Huge generuje wpis w RSS przy nawet najmniejszej zmianie treści wpisu.

**Reply by spexpl** [reply]
Po ostatnim wpisie, chyba doszedłem czemu Hugo czasem republikuje stare wpisy na RSSie. Winą jest linkowanie wewnętrzne. W ostatnim wpisie, wstawiłeś dwa odnośniki do postu o X i 40 lvl, w efekcie czego Hugo republikował oba wpisy.

**Reply by idarek** [reply]
Nie do końca. 

Opublikowałem artykuł po angielsku, który ma odnośniki do 2 innych wpisów które tylko miałem po angielsku na stronie. Jako że staram się tłumaczyć nowe wpisy, żeby miało sens, musiałem przetłumaczyć 2 wpisy z grudnia 2022 aby odnośniki były do poprawnych wersji językowych. Jako że te przetłumaczone strony to tak naprawdę nowe odnośniki, które nie były jeszcze w Internecie i w RSS, to czytniki RSS również wyłapią to, że to są nowe wpisy, bez znaczenia że mają one tą samą datę publikacji, mianowicie grudzień 2022.

**Reply by spexpl** [reply]
To jak Hugo ogarnia różne wersje językowe wpisów? Bo jak dla mnie to wpis PL i EN to są dwa różne wpisy.

**Reply by idarek** [reply]
Dwa są pisane niezależnie a łączę je w Hugo przy pomocy `translationKey`.

John Mueller potwierdził jakiś czas temu, że tłumaczone wpisy nie koniecznie muszą zawierać te same informacje.
